src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -5429,60 +5445,114 @@ def initialize(self):
 
     # --------------------------------------------------------------------------
     #
-    def finalize(self):
+    def alive_barrier(self):
+
+        # FIXME: wait for bridges, too?  But we need pubsub for counting... Duh!
+        total = len(self._bridges)    + \
+                len(self._components) + \
+                len(self._workers   ) + \
+                len(self._sub_agents)
+        start   = time.time()
+        timeout = 60
```

Comment:
Well, it still does make  sure we are not caught pants down with no bridges: if the components come up, but the bridges not (or can't be used because of misconfiguration), then we won't get an alive message.  Which is good, because we would not get any other messages  either, such as units.  The timeout will prevent that scenario from resulting in a deadlock.  I don't think it serves any other purpose right now, but that is good enough until we have some more active bridge health monitoring (if ever).